### DISCUSSION TITLE
Disable VAT/Taxes on specific cases

### DISCUSSION THREAD

**Discussion by Web10-Joris** [duncanmcclean/simple-commerce#989] [Help] [answered]
Is it possible to overwrite the vat/tax rules in some order cases? I want to add a field to each order with company and VAT ID, then when the VAT ID is valid (check via external API) the VAT should be disabled for this order and recalculate the totals. What is the best approach for this? 

```
if($vatId->valid) {
        Clear out the previous calculated default VAT for this order
    }
    else {
        Continue as usual, including the normal VAT
    }
```

**Comment by duncanmcclean** [comment]
Hey 👋

Which tax engine are you using? Basic or Standard? (it'll tell you if you run `php please support:details`)

**Reply by Web10-Joris** [reply]
Hi, oops forgot to mention. I'm using the Standard Tax Engine.

**Comment by duncanmcclean** [comment, marked as answer]
I believe you should be able to achieve this by creating your own tax engine, which extends the "Standard Tax Engine" you're already using.

You first need to create your tax engine class - I've just put mine in `app` and called it `CustomTaxEngine`, feel free to call it whatever you want.

The `calculateForLineItem` method is responsible for calculating the tax for each line item. 

As a really basic example, I've got the `$hasValidVat` variable which, when `true`, will return a 0% tax rate for the line item. Otherwise, it'll fallback to the normal standard tax engine logic. You can obviously change this as needed.

```php
<?php

namespace App;

use DuncanMcClean\SimpleCommerce\Contracts\Order;
use DuncanMcClean\SimpleCommerce\Orders\LineItem;
use DuncanMcClean\SimpleCommerce\Tax\TaxCalculation;

class CustomTaxEngine extends \DuncanMcClean\SimpleCommerce\Tax\Standard\TaxEngine
{
    public function calculateForLineItem(Order $order, LineItem $lineItem): TaxCalculation
    {
        $hasValidVat = true;

        if ($hasValidVat) {
            return new TaxCalculation(0, 0, false);
        }

        return parent::calculateForLineItem($order, $lineItem);
    }
}
```

Then, all you need to do is swap over the `tax_engine` in your config file to your new custom one:

```php
// config/simple-commerce.php

'tax_engine' => \App\CustomTaxEngine::class,
```

_^ btw, I've just tagged v6 so you may need to replace `DuncanMcClean` in the imports to `DoubleThreeDigital` if you're still using v5._

**Reply by Web10-Joris** [reply]
Hi @duncanmcclean, many thanks for this comprehensive response! Sounds like a perfect solution. I am going to try this immediately after upgrading from v5 to v6.
